catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -465,6 +474,23 @@ private void skipBytes(InputStream is, String bytesToSkip) throws IOException {
         }
     }
 
+    private void alignStream(InputStream in, long requestedBytesToSkip, long responseBytesSkipped)
+            throws IOException {
+        try {
+            if (requestedBytesToSkip > responseBytesSkipped) {
+                LOGGER.error(
+                        "Source incorrect byte range, skipping ahead [{}] bytes in the input stream",
+                        requestedBytesToSkip - responseBytesSkipped);
+                long skipped = in.skip(requestedBytesToSkip - responseBytesSkipped);
+            } else if (requestedBytesToSkip < responseBytesSkipped) {
+                throw new IOException(
+                        "Source server skipped more bytes than requested in the range header.");
+            }
+        } catch (IOException e) {
```

Comment:
Same comment as previous. I'll log an error as well, but it should be ok to throw an IOException and be treated as a failed download attempt.